### PR TITLE
Refactor search components 833

### LINKED ISSUE #833: Refactor Producer Search and Node Search component
See the logic for Search Producer and Nodes and create a common component for search input and refactoring those components using the new one

### PR DESCRIPTION
**Pull request by Torresmorah**
### What does this PR do?

- Resolve #833 
- Add new component for search based on NodeSearch and ProducerSearch components 
- Use the new component in block  producers, nodes, geo distribution and accounts pages
- Change the port of hasura in env files

### Steps to test

1. Run the project locally
2. Test the search bar components in the changed pages

#### CheckList

- [ ] The content is adequate
- [ ] The content is available in both English and Spanish
- [ ] I Ran a spell check

### CODE REVIEW

File: webapp/src/routes/Accounts/index.js [hunk 1]
```diff
@@ -103,6 +99,8 @@ const Accounts = ({ ual }) => {
   }
 
   const handleOnSearch = async (valueAccount) => {
+    const accountName = valueAccount?.owner ?? ''
```
Comment (tetogomez) [inline webapp/src/routes/Accounts/index.js:102]: Nullish Coalescing, nice bro!! very nice! 😂

File: webapp/src/routes/Accounts/index.js [hunk 2]
```diff
@@ -112,7 +110,7 @@ const Accounts = ({ ual }) => {
     await new Promise((resolve) => setTimeout(resolve, 500))
 
     try {
-      const account = await eosApi.getAccount(valueAccount)
+      const account = await eosApi.getAccount(accountName)
```
Comment (tetogomez) [inline webapp/src/routes/Accounts/index.js:113]: add blank space after const declaration

File: webapp/src/routes/Accounts/index.js [hunk 3]
```diff
@@ -122,9 +120,9 @@ const Accounts = ({ ual }) => {
     }
 
     try {
-      const { abi } = await eosApi.getAbi(valueAccount)
+      const { abi } = await eosApi.getAbi(accountName)
       setAbi(abi)
-      const { code_hash: hash = '' } = await eosApi.getCodeHash(valueAccount)
+      const { code_hash: hash = '' } = await eosApi.getCodeHash(accountName)
```
Comment (tetogomez) [inline webapp/src/routes/Accounts/index.js:125]: same here!

File: webapp/src/routes/Accounts/index.js [hunk 4]
```diff
@@ -133,60 +131,34 @@ const Accounts = ({ ual }) => {
...
+      handleOnSearch({ owner: 'eosio' })
 
       return
     }
 
-    setAccountName(params.account)
-    handleOnSearch(params.account)
+    setFilters({ owner: params.account })
+    handleOnSearch({ owner: params.account })
     // eslint-disable-next-line
```
Comment (tetogomez) [inline webapp/src/routes/Accounts/index.js:146]: we can fix this eslint comment wrapping `handleOnSearch` function with useCallback, please take a look!

File: webapp/src/routes/Accounts/index.js [hunk 5]
```diff
@@ -133,60 +131,34 @@ const Accounts = ({ ual }) => {
...
   }, [location.search])
 
   useEffect(() => {
-    handleOnSearch('eosio')
+    handleOnSearch({ owner: 'eosio' })
     // eslint-disable-next-line
```
Comment (tetogomez) [inline webapp/src/routes/Accounts/index.js:151]: same here!

File: webapp/src/routes/BlockProducers/index.js [hunk 6]
```diff
@@ -60,11 +63,14 @@ PaginationWrapper.propTypes = {
   totalPages: PropTypes.number,
   page: PropTypes.number,
   onPageChange: PropTypes.func,
-  loading: PropTypes.bool,
-  chipFilter: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+  loading: PropTypes.bool
 }
 
 const Producers = () => {
+  const chips = CHIPS_NAMES.map((e) => {
```
Comment (tetogomez) [inline webapp/src/routes/BlockProducers/index.js:70]: we can do that as a single line function

example: 
` const chips = CHIPS_NAMES.map((e) => ({ name: e }))`

File: webapp/src/routes/BlockProducers/index.js [hunk 7]
```diff
@@ -174,22 +179,22 @@ const Producers = () => {
   }, [dataHistory, loadingHistory])
 
   return (
-    <Box>
+    <Grid>
```
Comment (tetogomez) [inline webapp/src/routes/BlockProducers/index.js:182]: if you won't be using the real potential of the `Grid` or `Box` component is much better just use a simple `div` as a wrapper

File: webapp/src/routes/Nodes/index.js [hunk 8]
```diff
@@ -146,8 +163,13 @@ const Nodes = () => {
   }, [filters, producers])
 
   return (
-    <Box>
-      <NodeSearch filters={filters} onChange={handleOnFiltersChange} />
+    <Grid>
```
Comment (tetogomez) [inline webapp/src/routes/Nodes/index.js:166]: same here

File: webapp/src/routes/NodesDistribution/index.js [hunk 9]
```diff
@@ -97,25 +115,24 @@ const Nodes = () => {
...
     }
 
     setNodes(items)
   }, [allNodes, filters])
 
   return (
-    <Box>
-      <NodeSearch
-        producers={producers}
+    <Grid>
```
Comment (tetogomez) [inline webapp/src/routes/NodesDistribution/index.js:126]: and here!

**Review by tetogomez — CHANGES_REQUESTED** [review]
Please check the comment @Torresmorah 


Also, you touch several components and routes that you can implement the new logic for single responsibility, let me know if U need support for it

**Review by AngeloCG97 — APPROVED** [review]
LGTM

**Review by fvives99 — APPROVED** [review]
LGTM

File: webapp/src/hooks/customHooks/useSearchState.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+const useSearchState = ({ query }) => {
+  const [loadProducers, { loading = true, data: { producers, info } = {} }] =
+    useLazyQuery(query)
+  const location = useLocation()
+  const [pagination, setPagination] = useState({ page: 1, pages: 1, limit: 28 })
+  const [filters, setFilters] = useState({ name: 'all', owner: '' })
+
+  const handleOnSearch = (newFilters) => {
+
```
Comment (tetogomez) [inline webapp/src/hooks/customHooks/useSearchState.js]: delete this blank space

**Review by tetogomez — APPROVED** [review]
LGTM

**Pull request merged**